src/Units/UnitModel.java

```diff
@@ -189,12 +193,40 @@ public boolean canRepairOn(Location locus)
   }
 
   /** Provides a hook for inheritors to supply turn-initialization actions to a unit.
+   * Overriders should collect the output of super.getTurnInitEvents() before returning.
    * @param self Assumed to be a Unit of the model's type.
+   * @param map The current true state of the game map.
    */
   public GameEventQueue getTurnInitEvents(Unit self, MapMaster map)
   {
-    // Most Units don't have any; specific UnitModel types can override.
-    return new GameEventQueue();
+    GameEventQueue queue = new GameEventQueue();
+    if( isAirUnit() && 0 == self.fuel )
+    {
+      // Uh oh. We are about to go crashy crashy. Unless we are about to be resupplied...
+      boolean ok = false;
+      XYCoord xyc = new XYCoord(self.x, self.y);
+      Location loc = map.getLocation(xyc);
+
+      // First check if we are on an airport.
+      if(loc.getOwner() == self.CO && canRepairOn(loc))
+        ok = true;
+
+      // Next see if there is a friendly APC nearby.
+      List<XYCoord> adjacents = Utils.findLocationsInRange(map, xyc, 1);
+      for( XYCoord adj : adjacents )
+      {
+        Unit res = map.getLocation(adj).getResident();
+        if( null != res && (res.CO == self.CO) && res.model.hasActionType(UnitActionFactory.RESUPPLY) )
```

Comment:
May as well add the auto-resupply logic into this function, since you're now depending on units having the `RESUPPLY` action == doing auto-supply on turn init.
I'm not opposed to that notion, mind, it's just not a constraint we currently enforce and you're touching the logic.

Reply:
It would be a little strange to move that here, though I guess there's no reason we can't. If we ever make units that can resupply one thing but not others, we'll have issues anyway.
I can pull that in if you want. Seems odd since I think APCs are what made us create the `UnitModel` event hook in the first place.

Reply:
> It would be a little strange to move that here, though I guess there's no reason we can't.

It'd be even weirder to have a supplying unit that *can't* auto-supply be able to keep an empty jet in the air by its mere presence.

> If we ever make units that can resupply one thing but not others, we'll have issues anyway.

For your reference, then, in case we ever go for a SFW unit set:
Fuel Trucks in Super Famicom Wars do have auto-supply, but cannot supply jets.
I don't recall at the moment whether they can resupply ships.
In addition, they can only resupply copters on open ground (IIRC only forests and mountains are banned. If it's relevant, I believe it's 1:1 with the places T-Copters can drop units from.)

> I can pull that in if you want.

I would prefer that.

> Seems odd since I think APCs are what made us create the `UnitModel` event hook in the first place.

They were.
You're depending on that behavior now, though, so IMO it goes here unless and until we have a reason to change/"better" solution.
The first thing to mind would be an "auto-supply" unit role.

Reply:
I almost created a SUPPLY unit role, and then didn't because I realized it would be duplicative of "has the RESUPPLY action"